### PR TITLE
Refactor Team Assessment API

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Created a `staff_base` file to house some shared logic between team and staff assessment APIs with the goal of reducing duplicate code.

**What changed?**
- New `staff_base` file
- Extended `Team/StaffWorkflow` to allow for getting by `identifying_uuid`
- Genericized workflow cancellations and get of latest assessment
- Move atomic assessment creation code into `staff_base`

**Notes**
- For the sake of time I haven't given **every** function a generic, shared version. There's more refactoring that can be done, but this change set includes some quick-wins.
- Unlike the pre-refactor code, this touches staff assessment code which increases risk of regression.

FYI @edx/masters-devs-gta 
Related to #1393

### CODE REVIEW

**Comment by nsprenkle** [comment]
Closing as not having reached stable-state before end of time box.
